typed-racket-lib/typed-racket/typecheck/tc-send.rkt

```diff
@@ -58,18 +58,19 @@
 ;; Handles typechecking the actual application inside the method send
 ;; expansion. Most of the work is done by tc/app via tc-expr.
 (define (tc/send-internal vars types app-stx expected)
+  (define bindings (map -binding vars types))
```

Comment:
I worry about about all the extra list allocations. Do we need to represent these as lists of pairs?